test/functional/feature_reindex.py

```diff
@@ -97,6 +121,77 @@ def continue_reindex_after_shutdown(self):
             node.wait_for_rpc_connection(wait_for_import=False)
         node.stop_node()
 
+    # Check that reindex uses -assumevalid
+    def assume_valid(self):
+        self.log.info("Testing reindex with -assumevalid")
+        self.start_node(0)
+        node = self.nodes[0]
+        self.generate(node, 1)
+        coinbase_to_spend = node.getblock(node.getbestblockhash())['tx'][0]
+        # Generate COINBASE_MATURITY blocks to make the coinbase spendable
+        self.generate(node, COINBASE_MATURITY)
+        tip = int(node.getbestblockhash(), 16)
+        block_info = node.getblock(node.getbestblockhash())
+        block_time = block_info['time'] + 1
+        height = block_info['height'] + 1
+        blocks = []
+
+        # Create a block with an invalid signature
+        tx = CTransaction()
+        tx.vin.append(CTxIn(COutPoint(int(coinbase_to_spend, 16), 0), scriptSig=b""))
+        tx.vout.append(CTxOut(49 * 10000, CScript([OP_TRUE])))
+        tx.calc_sha256()
+        invalid_block = create_block(tip, create_coinbase(height), block_time, txlist=[tx])
+        invalid_block.solve()
+        tip = invalid_block.sha256
+        blocks.append(invalid_block)
+
+        # Bury that block with roughly two weeks' worth of blocks (2100 blocks) and an extra 100 blocks
+        for _ in range(2200):
```

Comment:
is there a reason we need to mine that many blocks, and not just a few more to make the test faster?

Reply:
Are you referring to the 100 blocks mined after the first 2100 blocks?

Reply:
I meant the 2100 (two weeks) of of blocks. I didn't test this, but is there a reason this couldn't just be a few blocks (plus a few more that aren't sent to the node)?

Reply:
Script checks are only skipped when the current block is roughly 2 weeks away from the tip. See https://github.com/bitcoin/bitcoin/blob/3f83c744ac28b700090e15b5dda2260724a56f49/src/validation.cpp#L2476-L2494

We have to bury the test block 2 weeks deep for script verification to the skipped

Reply:
oh I see, forgot about that check - can be resolved.

Reply:
@stratospher the test completes much faster on my system, but by checking the timing of other tests, I put it among "Tests less than 2m" section because it completes faster than `mining_getblocktemplate_longpoll.py` than slower than `mempool_persist.py`